### PR TITLE
Chore/move nft ext trait

### PR DESCRIPTION
**Pull request by JasonTulp**
Migrates NFTExt trait from pallet_nft to seed_pallet_common
Moves get_collection_info into it's own trait (That remains in NFT pallet)

### CODE REVIEW

File: pallet/doughnut/src/test.rs [hunk 1]
```diff
@@ -807,6 +807,7 @@ fn generate_alice_to_bob_outer_signature_for_balances_transfer_keep_alive() {
 			let mut outer_call_payload: Vec<u8> = outer_call.encode();
 			outer_call_payload.as_mut_slice()[1] = 0x05; // due to real runtime pallet versioning
 			let outer_signature = holder.sign_ecdsa(&outer_call_payload.as_slice());
+			// println!("outer signature: {:?}", to_hex(&outer_signature, false));
```
Comment (zees-dev) [inline pallet/doughnut/src/test.rs:810]: remove this change; probably accidental

File: pallet/marketplace/Cargo.toml [hunk 2]
```diff
@@ -21,8 +21,8 @@ scale-info = { version = "2.3.0", default-features = false, features = ["derive"
 
 seed-primitives = { path = "../../primitives", default-features = false }
 seed-pallet-common = { path = "../common", default-features = false }
-pallet-nft = { path = "../nft", default-features = false }
-pallet-sft = { path = "../sft", default-features = false }
+pallet-nft = { path = "../nft", default-features = false, features = ["frame-benchmarking"] }
+pallet-sft = { path = "../sft", default-features = false, features = ["frame-benchmarking"] }
```
Comment (zees-dev) [inline pallet/marketplace/Cargo.toml:25]: wait why we need the frame-benchmarking feature for prod builds

File: pallet/nft/src/traits.rs [hunk 3]
```diff
@@ -14,74 +14,21 @@
...
-use sp_runtime::{BoundedVec, DispatchError, DispatchResult};
-use sp_std::{fmt::Debug, vec::Vec};
+use seed_primitives::CollectionUuid;
+use sp_runtime::DispatchError;
+use sp_std::fmt::Debug;
 
 use crate::CollectionInformation;
 
-pub trait NFTExt {
+pub trait NFTCollectionInfo {
```
Comment (zees-dev) [inline pallet/nft/src/traits.rs:23]: nit: not sure if we even need the trait - since anything that uses this trait can use the function directly

**Review by zees-dev — APPROVED** [review]
Few minor issues; otherwise LGTM 👍

**Review by surangap — APPROVED** [review]
lgtm. would be interested to know why we need this -> https://github.com/futureversecom/trn-seed/pull/789/files#r1499077162

File: pallet/marketplace/Cargo.toml (hunk 2 above)
Comment (zees-dev) [reply pallet/marketplace/Cargo.toml]: addressed: https://github.com/futureversecom/trn-seed/pull/789/commits/113ac390effeca8f4bab0cb9c701141d25384c2b

File: pallet/doughnut/src/test.rs (hunk 1 above)
Comment (zees-dev) [reply pallet/doughnut/src/test.rs]: addressed: https://github.com/futureversecom/trn-seed/pull/789/commits/e8728bc222f1c3106c1f341ce9776c307294bd21

**Pull request merged**